apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -146,7 +146,13 @@ where
                     if !cfg!(feature = "ABCI") {
                         self.storage.tx_queue.pop();
                     }
-                    Event::new_tx_event(&tx_type, height.0)
+                    let mut event = Event::new_tx_event(&tx_type, height.0);
+                    if let DecryptedTx::Undecryptable(_) = inner {
+                        event["info"] =
+                            "Transaction could not be decrypted. ".into();
+                        event["code"] = ErrorCodes::Undecryptable.into();
```

Comment:
I'm a little confused about this. When the `inner` is `Undecryptable`, shouldn't we also `continue` the loop? It looks like it still goes on to apply it below

Reply:
oh, it's because it's not yet being used, right?

Reply:
I see, thank you!